devito/passes/iet/petsc.py

```diff
@@ -0,0 +1,120 @@
+from devito.passes.iet.engine import iet_pass
+from devito.ir.iet import (Expression, FindNodes, Section, List,
+                           Callable, Call, Transformer, Callback,
+                           Definition, Uxreplace, FindSymbols)
+from devito.ir.equations.equation import OpAction
+from devito.types.petsc import Mat, Vec, DM, PetscErrorCode, PETScStruct, PETScArray
+from devito.symbolics import FieldFromPointer, Byref
+
+
+__all__ = ['lower_petsc']
+
+
+@iet_pass
+def lower_petsc(iet, **kwargs):
+    # NOTE: Currently only considering the case when opt='noop'. This
+    # does not deal with the Temp Expressions generated when opt is not set
+    # to 'noop' etc.
+
+    # Find the Section containing the 'action'. For now, assume we're only solving
+    # 1 equation via PETSc so only 1 action exists within a single Section.
+    sections = FindNodes(Section).visit(iet)
```

Comment:
Why do you need to know about the Section? This seems weird and unnecessary at first